src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -60,8 +68,11 @@ static int peek_empty_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
 
     switch (MPIDI_OFI_REQUEST(rreq, event_id)) {
         case MPIDI_OFI_EVENT_PEEK:
-            MPIDI_OFI_REQUEST(rreq, util_id) = MPIDI_OFI_PEEK_NOT_FOUND;
             rreq->status.MPI_ERROR = MPI_SUCCESS;
+            /* util_id should be the last thing to change in rreq. Reason is
+             * we use util_id to indicate peek_event has completed and all the
+             * relevant values have been copied to rreq. */
```

Comment:
I don't think the logic of this comment is reliable. Unless I am misunderstanding, we shouldn't assume updates to struct members are ordered.